extensions/binding/org.eclipse.smarthome.binding.coapconnectedlighting/src/main/java/org/eclipse/smarthome/binding/coapconnectedlighting/handler/CoAPConnectedLightingHandler.java

```diff
@@ -0,0 +1,361 @@
+
+import static org.eclipse.californium.core.coap.MediaTypeRegistry.TEXT_PLAIN;
+import static org.eclipse.smarthome.binding.coapconnectedlighting.CoAPConnectedLightingBindingConstants.*;
+
+import java.math.BigDecimal;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.californium.core.CoapClient;
+import org.eclipse.californium.core.CoapResponse;
+import org.eclipse.californium.core.WebLink;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.IncreaseDecreaseType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link CoAPConnectedLightingHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Microchip Technology - I nitial contribution
+ */
+public class CoAPConnectedLightingHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(CoAPConnectedLightingHandler.class);
+
+    private ChannelUID stringChannelTemperatureUID;
+    private ChannelUID stringChannelHumidityUID;
+    private ChannelUID stringChannelPressureUID;
+
+    ScheduledFuture<?> refreshTemperatureJob;
+    ScheduledFuture<?> refreshHumidityJob;
+    ScheduledFuture<?> refreshPressureJob;
+
+    private BigDecimal refresh;
+
+    CoapClient coapClient = new CoapClient();
+
+    String COAP_SERVER_IP_ADDRESS = "0.0.0.0";
+    int COAP_SERVER_PORT = 5683;
+
+    /*
+     * String COAP_SERVER_IP_ADDRESS_LIGHT = "192.168.1.103";
+     * String COAP_SERVER_PORT_LIGHT = "5683";
+     */
+    public CoAPConnectedLightingHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+
+        // Device Configuration Check
+        String deviceIPAddress = (String) this.getConfig().get("ipAddress");
+        int devicePortNumber = -1;
+
+        if (((BigDecimal) this.getConfig().get("portNumber")) != null) {
+            devicePortNumber = ((BigDecimal) this.getConfig().get("portNumber")).intValue();
+        }
+
+        if ((deviceIPAddress != null) && (devicePortNumber != -1)) {
+            if ((!deviceIPAddress.isEmpty()) && (devicePortNumber != -1)) {
+                COAP_SERVER_IP_ADDRESS = deviceIPAddress;
+                COAP_SERVER_PORT = devicePortNumber;
+            } else {
```

Comment:
You may turn this around and check the error path in one condition (since the resulting status and logger entries are identical) like this:

``` 
if (StringUtils.isBlank(deviceIPAddress) || devicePortNumber == -1) {
    logger.debug(this.getThing().getUID() + "-Invalid IP Address or Port Number");
    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
                    "Can not access device as the ip address or port number is invalid");
    return;
}
        
COAP_SERVER_IP_ADDRESS = deviceIPAddress;
COAP_SERVER_PORT = devicePortNumber;
```

note: for StringUtils you need to define a separate import package to apache.commons.